src/main/java/com/stuypulse/robot/subsystems/Edwin.java

```diff
@@ -0,0 +1,229 @@
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+
+        leftTargetSpeed = 0;
+        rightTargetSpeed = 0;
+
+        odometry = new DifferentialDriveOdometry(getRotation2d());
+        kinematics = new DifferentialDriveKinematics(EdwinSettings.TRACK_WIDTH);
+        navx = new AHRS(SPI.Port.kMXP);
+
+        right.setInverted(true);
+        reset(EdwinSettings.Odometry.STARTING_POSITION);
+
+        field = new Field2d();
+    }
+
+    /*********************
+     * ENCODER FUNCTIONS *
+     *********************/
+
+    private void setGreyhillDistancePerPulse(double distance) {
+        rightGreyhill.setDistancePerPulse(distance);
+        rightGreyhill.reset();
+
+        leftGreyhill.setDistancePerPulse(distance);
+        leftGreyhill.reset();
+    }
+
+    // Distance
+    private double getLeftDistance() {
+        return leftGreyhill.getDistance();
+    }
+
+    private double getRightDistance() {
+        return rightGreyhill.getDistance();
+    }
+
+    private double getDistance() {
+        return (getLeftDistance() + getRightDistance()) / 2.0;
+    }
+
+    // Velocity
+    private double getLeftVelocity() {
+        return leftGreyhill.getRate();
+    }
+
+    private double getRightVelocity() {
+        return rightGreyhill.getRate();
+    }
+
+    private double getVelocity() {
+        return (getLeftVelocity() + getRightVelocity()) / 2.0;
+    }
+
+    /***************
+     * ROBOT ANGLE *
+     ***************/
+
+    // Gets current Angle of the Robot as a double (contiuous / not +-180)
+    public double getRawGyroAngle() {
+        return navx.getAngle();
+    }
+
+    // Gets current Angle of the Robot
+    public Angle getGyroAngle() {
+        return Angle.fromDegrees(getRawGyroAngle());
+    }
+
+    /**********************
+     * ODOMETRY FUNCTIONS *
+     **********************/
+
+    private void updateOdometry() {
+        odometry.update(getRotation2d(), getLeftDistance(), getRightDistance());
+    }
+
+    public Rotation2d getRotation2d() {
+        // TODO: check if this needs to be negative
+        return getGyroAngle().negative().getRotation2d();
+    }
+
+    public Pose2d getPose() {
+        updateOdometry();
+        return odometry.getPoseMeters();
+    }
+
+    @Override
+    public void setPose(Pose2d pose) {
+        reset();
+
+        odometry.resetPosition(pose, new Rotation2d());
+    }
+
+    @Override
+    public DifferentialDriveKinematics getKinematics() {
+        return kinematics;
+    }
+
+    @Override
+    public TrajectoryConfig getTrajectoryConfig() {
+        return new TrajectoryConfig(
+            EdwinSettings.Motion.MAX_VELOCITY, 
+            EdwinSettings.Motion.MAX_ACCELERATION
+        ).setKinematics(kinematics);
+    }
+
+    /****************
+     * SENSOR RESET *
+     ****************/
+
+    public void reset(Pose2d location) {
+        leftGreyhill.reset();
+        rightGreyhill.reset();
+
```

Comment:
rename this setPose(Pose2d) and remove everything else pretaining to resetting.. this is the only needed / correct thing